src/adc_dma.cpp

```diff
@@ -76,6 +76,51 @@ void calibrate_current_sensors()
     load_current_offset = -(float)(((adc_filtered[ADC_POS_I_LOAD] >> (4 + ADC_FILTER_CONST)) * vcc) / 4096) * ADC_GAIN_I_LOAD / 1000.0;
 }
 
+void adc_update_value(unsigned int pos)
+{
+    // low pass filter with filter constant c = 1/16
+    // y(n) = c * x(n) + (c - 1) * y(n-1)
+
+#if FEATURE_PWM_SWITCH == 1
+    if (pos == ADC_POS_V_SOLAR || pos == ADC_POS_I_SOLAR) {
+        // only read input voltage and current when switch is on or permanently off
+        if (pwm_switch.signal_high() || pwm_switch.active() == false) {
+            adc_filtered[pos] += (uint32_t)adc_readings[pos] - (adc_filtered[pos] >> ADC_FILTER_CONST);
+        }
+    }
+    else
+#endif
+    {
+        // adc_readings: 12-bit ADC values left-aligned in uint16_t
+        adc_filtered[pos] += (uint32_t)adc_readings[pos] - (adc_filtered[pos] >> ADC_FILTER_CONST);
+    }
+
+    // check if limit violation alert callback is set for this ADC channel
+    if (adc_alerts[pos].callback != NULL) {
+
+        uint32_t err_flag = 0;
+        if (adc_readings[pos] > adc_alerts[pos].upper_limit) {
+            err_flag |= adc_alerts[pos].err_flag_upper;
+            adc_alerts[pos].debounce_counter++;
+        }
+        else if (adc_readings[pos] < adc_alerts[pos].lower_limit) {
+            err_flag |= adc_alerts[pos].err_flag_lower;
+            adc_alerts[pos].debounce_counter++;
+        }
+        else {
+            adc_alerts[pos].debounce_counter = 0;
+        }
+
+        // minimum debouncing that a single spike on ADC measurement doesn't trigger the alert
+        if (adc_alerts[pos].debounce_counter > 1) {
```

Comment:
This corresponds to 2 milliseconds right? Then I would mention this here in a comment.
I think we need to test if a 10msec check is better to avoid false triggering in case of battery loose contacts or similar.

Reply:
Well, both approaches have pro's and cons. An high frequency oscillation / ringing which in average is higher than the OV limit  and the period is shorter than the averaging time will be correctly detected by the average approach, not by the debouncer ( if debounce restart happens with each going within the limits and out again).